File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/world/WorldDirectory.java
package top.likoslupus.cellulosesz.api.world;

import java.util.List;
import java.util.Optional;

public interface WorldDirectory {

    List<String> loadedWorldIds();

    default Optional<String> resolveLoadedWorld(String input) {
        return resolve(input).worldId();
    }

    WorldResolution resolve(String input);

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/world/WorldDirectory.kt
package top.likoslupus.cellulosesz.api.world

public interface WorldDirectory {

    public fun loadedWorldIds(): List<String>

    public fun resolveLoadedWorld(input: String): String? {
        val resolution = resolve(input)
        return if (resolution is WorldResolution.Resolved) {
            resolution.worldId
        } else {
            null
        }
    }

    public fun resolve(input: String): WorldResolution

}
